storage/tscache: TestIntervalSklConcurrentVsSequential/useClock=true failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/46c3f81a8c396610e11d184cfc89b8b8dfb93079

Parameters:
```
TAGS=
GOFLAGS=-race
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=463862&tab=buildLog

```


ERROR: signal: killed

context canceled
make: *** [stress] Error 1
1 runs completed, 1 failures, over 8m56s
Makefile:798: recipe for target 'stress' failed
```

--- a/pkg/storage/tscache/cache_test.go
+++ b/pkg/storage/tscache/cache_test.go
@@ -555,8 +555,12 @@ func TestTimestampCacheImplsIdentical(t *testing.T) {
 				txnID := uuid.MakeV4()
 				maxVal := cacheValue{}
 
-				const n = 1000
-				for j := 0; j < n; j++ {
+				rounds := 1000
+				if util.RaceEnabled {
+					// Reduce the number of rounds for race builds.
+					rounds /= 2
+				}
+				for j := 0; j < rounds; j++ {
 					t.Logf("goroutine %d at iter %d", i, j)
 
 					// Wait for all goroutines to synchronize.

--- a/pkg/storage/tscache/interval_skl_test.go
+++ b/pkg/storage/tscache/interval_skl_test.go
@@ -951,8 +951,12 @@ func TestIntervalSklConcurrency(t *testing.T) {
 						txnID := uuid.MakeV4()
 						maxVal := cacheValue{}
 
-						const n = 1000
-						for j := 0; j < n; j++ {
+						rounds := 1000
+						if util.RaceEnabled {
+							// Reduce the number of rounds for race builds.
+							rounds /= 2
+						}
+						for j := 0; j < rounds; j++ {
 							// Choose a random range.
 							from, middle, to := randRange(rng, slots)
 							opt := randRangeOpt(rng)
@@ -1036,8 +1040,12 @@ func TestIntervalSklConcurrentVsSequential(t *testing.T) {
 			txnIDs[i] = uuid.MakeV4()
 		}
 
-		const n = 1000
-		for j := 0; j < n; j++ {
+		rounds := 1000
+		if util.RaceEnabled {
+			// Reduce the number of rounds for race builds.
+			rounds /= 2
+		}
+		for j := 0; j < rounds; j++ {
 			t.Logf("round %d", j)
 
 			// Create a set of actions to perform.

